### PR TITLE
601 Radio Option Unique Id Prop

### PR DESCRIPTION
**Pull request by pdavies88**
## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/601

## Testing done
Screen Shot 2022-01-28 at 12 24 56 PM

## Screenshots
Screen Shot 2022-01-28 at 12 28 39 PM
Screen Shot 2022-01-28 at 12 30 39 PM

## Acceptance criteria
- [x] va-radio component is updated to accept an optional id prop (with the label prop used as a default)
- [x] tests are updated accordingly
- [x] documentation is updated with proper usage for this prop (design.va.gov and Storybook)

## Definition of done
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by pdavies88** [comment]
@k80bowman we are still referencing the React Component in our docs on https://design.va.gov/components/form-controls#radio-buttons however this ticket was specifically a fix for the Web Component. 
The documentation in Storybook for the Web Component however is updated:
Screen Shot 2022-01-28 at 12 37 35 PM

**Comment by pdavies88** [comment]
@CourtneyChambers77 I don't happen to have Axe DevTools Pro so I'm not able to isolate elements for testing but if you have a Pro Suite would you be able to confirm that the Axe tests pass in regards to id's not being duplicated. You can test via this url: http://localhost:6006/iframe.html?id=components-va-radio--custom-id&viewMode=story

I'm confident it will be fine but I want to double check and make sure the issue reported originally via this screenshot in the ticket is resolved:
146273281-cac2903f-4558-47a3-bf8d-68237ea22a07

File: packages/web-components/src/components/va-radio-option/test/va-radio-option.e2e.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ describe('va-radio-option', () => {
   it('fires event for parent when changed', async () => {
     const page = await newE2EPage();
     await page.setContent(
-      '<va-radio-option aria-describedby="test" label="A label" value="something" />',
+      '<va-radio-option checked aria-describedby="test" label="A label" value="something" />',
```
Comment (bkjohnson) [inline packages/web-components/src/components/va-radio-option/test/va-radio-option.e2e.ts:26]: Is this related?

File: packages/web-components/src/components/va-radio-option/va-radio-option.tsx [hunk 2]
```diff
@@ -52,7 +57,7 @@ export class VaRadioOption {
         onClick={() => this.handleChange()}
         role="radio"
         value={this.value}
-        id={this.label}
+        id={this.customid || this.label}
```
Comment (bkjohnson) [inline packages/web-components/src/components/va-radio-option/va-radio-option.tsx:60]: Instead of giving people another prop that they have to think about, what if we just let them use `id`?

image

File: packages/web-components/src/components/va-radio-option/test/va-radio-option.e2e.ts (hunk 1 above)
Comment (pdavies88) [reply packages/web-components/src/components/va-radio-option/test/va-radio-option.e2e.ts:26]: @bkjohnson not directly but since I was messing with `va-radio` I figured I'd make this fix it solves this issue with us getting the `console.warn` during the `va-radio-option` tests
Prior to fix:
Screen Shot 2022-02-01 at 11 30 29 AM
Post fix:
Screen Shot 2022-02-01 at 11 32 08 AM

File: packages/web-components/src/components/va-radio-option/va-radio-option.tsx (hunk 2 above)
Comment (pdavies88) [reply packages/web-components/src/components/va-radio-option/va-radio-option.tsx]: @bkjohnson good call on this I have updated accordingly

File: packages/web-components/src/components/va-radio-option/va-radio-option.tsx [hunk 3]
```diff
@@ -52,9 +54,9 @@ export class VaRadioOption {
         onClick={() => this.handleChange()}
         role="radio"
         value={this.value}
-        id={this.label}
+        id={this.el.id || this.label}
       >
-        <label htmlFor={this.label}>{this.label}</label>
+        <label htmlFor={this.label} aria-label={this.name}>{this.label}</label>
```
Comment (bkjohnson) [inline packages/web-components/src/components/va-radio-option/va-radio-option.tsx:59]: Is this related to the `id` changes or is it a separate improvement? I'm not complaining about a11y enhancements, I'm just trying to get a better understanding.

**Review by bkjohnson — APPROVED** [review]

File: packages/web-components/src/components/va-radio-option/va-radio-option.tsx (hunk 3 above)
Comment (pdavies88) [reply packages/web-components/src/components/va-radio-option/va-radio-option.tsx:59]: This is due to Courtney's comment on having an accessibility error on the radio elements. I added her screen shots to the PR in the comment above

Comment (pdavies88) [reply packages/web-components/src/components/va-radio-option/va-radio-option.tsx]: Issue reported by @CourtneyChambers77 resolved via these steps: https://www.ssa.gov/accessibility/andi/help/alerts.html?no_name_generic

Screenshot to verify
Screen Shot 2022-02-03 at 12 06 20 PM

File: packages/web-components/src/components/va-radio-option/va-radio-option.tsx [hunk 4]
```diff
@@ -50,12 +52,13 @@ export class VaRadioOption {
       <Host
         aria-checked={this.checked ? `${this.checked}` : 'false'}
         aria-describedby={ariaDescribedby}
+        aria-label={this.name}
```
Comment (bkjohnson) [inline packages/web-components/src/components/va-radio-option/va-radio-option.tsx:55]: Can we get this covered in a test to try to prevent regressions?

Comment (pdavies88) [reply packages/web-components/src/components/va-radio-option/va-radio-option.tsx:55]: @bkjohnson updated code has been pushed to include a test covering this here: https://github.com/department-of-veterans-affairs/component-library/blob/ae05082e1bd712b70fc3d6fd773321e84cf6e48d/packages/web-components/src/components/va-radio-option/test/va-radio-option.e2e.ts#L12

**Comment by pdavies88** [comment]
Per @CourtneyChambers77 "Looks good with JAWS the SR works with it - Pass it"
